app/src/preload/index.ts

```diff
@@ -1,14 +1,45 @@
 // See the Electron documentation for details on how to use preload scripts:
 // https://www.electronjs.org/docs/latest/tutorial/process-model#preload-scripts
 import { contextBridge, ipcRenderer } from "electron";
-import type { ProxyRequest, SyncMetadataRequest } from "../types";
+import type {
+  ProxyRequest,
+  ProxyResponse,
+  SyncMetadataRequest,
+} from "../types";
+
+// Log the performance of IPC calls
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function logIpcCall<T>(name: string, fn: (...args: any[]) => Promise<T>) {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  return async (...args: any[]): Promise<T> => {
+    console.debug(`[IPC] ${name} called`);
+    const start = performance.now();
+    const result = await fn(...args);
+    const end = performance.now();
+    const duration = end - start;
+    let timeStr;
+    if (duration < 1000) {
+      timeStr = `${duration.toFixed(2)}ms`;
+    } else {
+      timeStr = `${(duration / 1000).toFixed(2)}s`;
+    }
+    console.debug(`[IPC] ${name} finished in ${timeStr}`);
+    return result;
+  };
+}
 
 const electronAPI = {
-  request: (request: ProxyRequest) => ipcRenderer.invoke("request", request),
-  requestStream: (request: ProxyRequest, downloadPath: string) =>
-    ipcRenderer.invoke("request", request, downloadPath),
-  syncMetadata: (request: SyncMetadataRequest) =>
+  request: logIpcCall<ProxyResponse>("request", (request: ProxyRequest) =>
```

Comment:
typescript is unhappy about this now: `Generic type 'ProxyResponse' requires 1 type argument(s)`

In the past we didn't specify the return type so it was okay...should I do `ProxyResponse<any>`?